### ISSUE TITLE
Why cookie name is case-insensitive

### ISSUE THREAD

**Issue by jamel** [netty/netty#5736]
I fall in such situation when HTTP-server (implemented with Netty) receives two cookies: ```session_id``` and ```Session_id```. But after parsing such header only one cookie is returned from method ```decode```. 

Example:
```java
Set<Cookie> cookies = ServerCookieDecoder.STRICT.decode("session_id=a; Session_id=b");
for (Cookie cookie: cookies) {
    System.out.println(cookie.name());
}
```
will print only ```session_id```.

Is this a bug? [RFC 6265](https://tools.ietf.org/search/rfc6265) does not state that cookie names must be case insensitive.

**Comment by slandelle** [comment]
Yep, looks like a bug to me. The issue lays in `io.netty.handler.codec.http.cookie.DefaultCookie#equals` implementation that compares names with `equalsIgnoreCase`.

Would you like to contribute a fix?

**Comment by jamel** [comment]
yes, I'd like

**Comment by jamel** [comment]
https://github.com/netty/netty/pull/5737

**Comment by slandelle** [comment]
Can be closed, as of #5737
